File: VB/ObtainUnderlyingData/ViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Xpf.PivotGrid
Imports System.Data
Imports System.Data.SQLite


Namespace HowToObtainUnderlyingData
	<POCOViewModel>
	Public Class ViewModel
		Public ReadOnly Property DataSource() As New DataTable()
		Protected Sub New()
			Dim connection = System.Configuration.ConfigurationManager.ConnectionStrings("nwind").ConnectionString
			Using conn As New SQLiteConnection("Data Source=C:\DataSources\nwind.db;")
				conn.Open()
				Dim command As New SQLiteCommand("Select * from SalesPerson", conn)
				Dim reader As SQLiteDataReader = command.ExecuteReader()
				DataSource.Load(reader)
				reader.Close()
			End Using
		End Sub
		Public Sub ShowDrillDownData(ByVal cellInfo As CellInfo)
			Me.GetService(Of IDialogService)().ShowDialog(MessageButton.OK, "Drill Down Results", cellInfo)
		End Sub
	End Class
End Namespace

File: CS/ObtainUnderlyingData/ViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.POCO;
using DevExpress.Xpf.PivotGrid;
using System.Data;
using System.Data.SQLite;

namespace HowToObtainUnderlyingData {
    [POCOViewModel]
    public class ViewModel {
        public DataTable DataSource { get; } = new DataTable();
        protected ViewModel() {
            using (SQLiteConnection conn = new SQLiteConnection(@"Data Source=nwind.db;")) {
                conn.Open();
                SQLiteCommand command = new SQLiteCommand("Select * from SalesPerson", conn);
                SQLiteDataReader reader = command.ExecuteReader();
                DataSource.Load(reader);
                reader.Close();
            }
        }
        public void ShowDrillDownData(CellInfo cellInfo) {
            this.GetService<IDialogService>().ShowDialog(MessageButton.OK, "Drill Down Results", cellInfo);
        }
    }
}
